Setup/UpgradeData.php

```diff
@@ -48,14 +81,51 @@ public function upgrade(ModuleDataSetupInterface $setup, ModuleContextInterface
             $oldProcessedPath = $this->customerCancelUrl->getOldDefaultUrl();
             $newProcessedPath = $this->customerCancelUrl->processValue('');
 
-            $tableName = $this->resourceConnection->getTableName('core_config_data');
-            $replace = ['value'=> $newProcessedPath];
-            $where = ['path = ?' => $path,'value = ?'=>$oldProcessedPath];
+            $tableName = $this->getCoreTable();
+            $replace = ['value' => $newProcessedPath];
+            $where = [self::PATH_EQUAL => $path, 'value = ?' => $oldProcessedPath];
             try {
-                $this->resourceConnection->getConnection()->update($tableName,$replace,$where);
+                $this->resourceConnection->getConnection()->update($tableName, $replace, $where);
             } catch (Exception $e) {
-                $this->logger->error('UpgradeData Exception : ',[$e->getMessage()]);
+                $this->logger->error('UpgradeData Exception : ', [$e->getMessage()]);
             }
         }
+        if (version_compare($version, '2.8.3', '<')) {
```

Comment:
could you extract this part into another upgrade class called by this one ?
Then you also will be able to extract `version_compare($version, '2.5.0', '<')` previous upgrade in another class.

And finally we will have one upgrade class by version (if necessary) that will be called by this `UpgradeData` main class...

WDYT ?

Reply:
it's a habit -  it's necessary to call upgrade by the main class but if you think it's more readable i think it's possible

<img width="933" alt="image" src="https://user-images.githubusercontent.com/97046219/181643861-d7ac3def-82b6-4da5-a185-080d8c3ec4af.png">

Reply:
forgot this.
we keep this for a rainy day